javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/config/DeclarativeConfigEarlyInitAgentConfig.java

```diff
@@ -0,0 +1,121 @@
+import io.opentelemetry.sdk.autoconfigure.internal.SpiHelper;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.DeclarativeConfiguration;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.SdkConfigProvider;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.OpenTelemetryConfigurationModel;
+import io.opentelemetry.sdk.resources.Resource;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * Agent config class that is only supposed to be used before the SDK (and {@link
+ * io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties}) is initialized.
+ */
+public final class DeclarativeConfigEarlyInitAgentConfig implements EarlyInitAgentConfig {
+  private final OpenTelemetryConfigurationModel configurationModel;
+  private final ConfigProperties declarativeConfigProperties;
+  private final SdkConfigProvider configProvider;
+
+  DeclarativeConfigEarlyInitAgentConfig(String configurationFile) {
+    this.configurationModel = loadConfigurationModel(configurationFile);
+    configProvider = SdkConfigProvider.create(configurationModel);
+    this.declarativeConfigProperties =
+        new DeclarativeConfigPropertiesBridge(
+            configProvider, this.configurationModel.getLogLevel());
+  }
+
+  @Override
+  public boolean isAgentEnabled() {
+    return !Objects.equals(configurationModel.getDisabled(), true);
+  }
+
+  @Nullable
+  @Override
+  public String getString(String propertyName) {
+    return declarativeConfigProperties.getString(propertyName);
+  }
+
+  @Override
+  public boolean getBoolean(String propertyName, boolean defaultValue) {
+    return declarativeConfigProperties.getBoolean(propertyName, defaultValue);
+  }
+
+  @Override
+  public int getInt(String propertyName, int defaultValue) {
+    return declarativeConfigProperties.getInt(propertyName, defaultValue);
+  }
+
+  @Override
+  public void logEarlyConfigErrorsIfAny() {
+    // todo
+  }
+
+  private static OpenTelemetryConfigurationModel loadConfigurationModel(String configurationFile) {
+    try (FileInputStream fis = new FileInputStream(configurationFile)) {
+      return DeclarativeConfiguration.parse(fis);
+    } catch (IOException e) {
+      throw new DeclarativeConfigException("unable to read " + configurationFile, e);
+    }
+  }
+
+  @Override
+  public AutoConfiguredOpenTelemetrySdk installOpenTelemetrySdk(ClassLoader extensionClassLoader) {
+    OpenTelemetrySdk sdk =
+        DeclarativeConfiguration.create(
+            this.configurationModel, SpiHelper.serviceComponentLoader(extensionClassLoader));
+    Runtime.getRuntime().addShutdownHook(new Thread(sdk::close));
+    GlobalOpenTelemetry.set(sdk);
```

Comment:
this is setting the global instance in the bootstrap classpath it seems - because it's not available in https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/methods/javaagent/src/declarativeConfigTest/java/io/opentelemetry/javaagent/instrumentation/methods/MethodTest.java